source/app/service-providers/commands/file-new.ts

```diff
@@ -72,24 +70,47 @@ export default class FileNew extends ZettlrCommand {
       dirpath = arg.path
       isFallbackDir = false
     } else if (openDirectory !== null && await this._app.fsal.isDir(openDirectory)) {
-      // Choose the openDirectory
+      // Explicitly provided directory
       dirpath = openDirectory
       isFallbackDir = false
     }
-
     // Make sure we have a filename and have the user confirm this if applicable
     // Also, if the user does not want to be prompted BUT we had to use the
     // fallback directory, we should also prompt the user as otherwise it would
     // be opaque to the user where the notes end up in.
+
     if (
       // No name provided, but the user does not want an auto-generated name
       (arg.name === undefined && !newFileDontPrompt) ||
       // The user does not wish to provide a filename, but we've had to revert
       // to a fallback directory
       (newFileDontPrompt && isFallbackDir)
+      
     ) {
-      // The user wishes/needs to confirm the filename and/or the directory
-      const chosenPath = await this._app.windows.saveFile(path.join(dirpath, generatedName))
+      let defaultFileName = generatedName
+      // Add the extension to the filename based on the selected type
+      switch (type) {
+        case 'json':
+          if (!defaultFileName.endsWith('.json')) {
+            defaultFileName += '.json'  // Ensure the default extension for JSON files
+          }
+          break
+        case 'yaml':
+          if (!defaultFileName.endsWith('.yaml')) {
```

Comment:
I would add a check if the name ends with `.yml` which is also a valid extension for YAML files. We can still default to `.yaml` below, that's absolutely fine.